src/core/server/execution_context/execution_context_container.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import type { KibanaServerExecutionContext } from './execution_context_service';
+import type { KibanaExecutionContext } from '../../types';
+
+// Switch to the standard Baggage header. blocked by
+// https://github.com/elastic/apm-agent-nodejs/issues/2102
+export const BAGGAGE_HEADER = 'x-kbn-context';
+
+export function getParentContextFrom(
+  headers: Record<string, string>
+): KibanaExecutionContext | undefined {
+  const header = headers[BAGGAGE_HEADER];
+  return parseHeader(header);
+}
+
+function parseHeader(header?: string): KibanaExecutionContext | undefined {
+  if (!header) return undefined;
+  try {
+    return JSON.parse(decodeURIComponent(header));
+  } catch (e) {
+    return undefined;
+  }
+}
+
+/**
+ * @public
+ */
+export interface IExecutionContextContainer {
+  toString(): string;
+  toJSON(): Readonly<KibanaServerExecutionContext>;
+}
+
+export class ExecutionContextContainer implements IExecutionContextContainer {
+  readonly #context: Readonly<KibanaServerExecutionContext>;
+  constructor(context: Readonly<KibanaServerExecutionContext>) {
+    this.#context = context;
+  }
+  toString(): string {
+    const ctx = this.#context;
+    const contextStringified = ctx.type && ctx.id ? `kibana:${ctx.type}:${ctx.id}` : '';
+    return contextStringified ? `${ctx.requestId};${contextStringified}` : ctx.requestId;
```

Comment:
Should we also enforce the max length here for the `baggage` header as we do on the client?

Reply:
yeah, I can

Reply:
done